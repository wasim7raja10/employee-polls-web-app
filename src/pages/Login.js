import { useState } from "react";
import { connect } from "react-redux";
import { Navigate } from "react-router-dom";
import { handleLoginAuthedUser } from "../actions/authedUser";

const Login = ({ dispatch, loggedIn }) => {
  const [credentials, setCredentials] = useState({
    username: "tylermcginnis",
    password: "abc321",
  });
  if (loggedIn) {
    const urlParams = new URLSearchParams(window.location.search);
    const redirectUrl = urlParams.get('redirectTo');
    return <Navigate to={redirectUrl ? redirectUrl : "/"} />;
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(handleLoginAuthedUser(credentials));
  }; 

  const handleLogin = (username, password) => {
    dispatch(handleLoginAuthedUser({ username, password }));
  };

  return (
    <div className=" flex justify-center flex-col items-center gap-6">
      <h1 className=" text-blue-400 text-4xl p-8 text-center">Login</h1>
      <div className="dropdown">
        <label data-testid="existing-user-label" tabIndex="0" className="btn m-1">Existing User</label>
        <ul tabIndex="0" className="dropdown-content menu p-2 shadow bg-base-100 rounded-box w-52">
          <li onClick={() => {
            handleLogin("sarahedo", "password123")
          }}><span>Sarah Edo</span></li>
          <li onClick={() => {
           handleLogin("tylermcginnis", "abc321")
          }}><span>Tyler McGinnis</span></li>
          <li onClick={() => {
            handleLogin("mtsamis", "xyz123")
          }}><span>Mike Tsamis</span></li>
          <li onClick={() => {
           handleLogin("zoshikanlu", "pass246")
          }}><span>Zenobia Oshikanlu</span></li>
        </ul>
      </div>
      {/* Login Form with username and password */}
      <form onSubmit={handleSubmit} className=" form-control w-full max-w-xs space-y-6">
        <div className=" space-y-2">
          <label data-testid="username-label" htmlFor="username">Username</label>
          <input data-testid="username-input" onChange={
            (e) => setCredentials({
              ...credentials,
              username: e.target.value
            })
          } value={credentials.username} className="input input-bordered w-full max-w-xs" type="text" id="username" name="username" />
        </div>

        <div className=" space-y-2">
          <label data-testid="password-label" htmlFor="password">Password</label>
          <input data-testid="password-input" onChange={
            (e) => setCredentials({
              ...credentials,
              password: e.target.value
            })
          } value={credentials.password} className="input input-bordered w-full max-w-xs" type="password" id="password" name="password" />
        </div>
        <button data-testid="submit-login" className="btn">Login</button>
      </form>
    </div>
  );
}

const mapStateToProps = ({ authUser }) => {
  return {
    loggedIn: !!authUser,
  };
}

export default connect(mapStateToProps)(Login);
