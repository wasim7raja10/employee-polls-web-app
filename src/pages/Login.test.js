import { configureStore } from "@reduxjs/toolkit";
import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import Login from "./Login";
import React from "react";
import reducer from "../reducers";

const store = configureStore({reducer})

describe("Login", () => {
  it("should render the component", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toBeDefined();
    expect(view).toMatchSnapshot();
  });

  it("should display all elements", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );
    
    const existingUserLabelElement = view.getByTestId("existing-user-label");
    const usernameLabelElement = view.getByTestId("username-label");
    const usernameInputElement = view.getByTestId("username-input");
    const passwordLabelElement = view.getByTestId("password-label");
    const passwordInputElement = view.getByTestId("password-input");
    const submitButtonElement = view.getByTestId("submit-login");

    expect(usernameLabelElement.textContent).toBe("Username");
    expect(passwordLabelElement.textContent).toBe("Password");
    expect(submitButtonElement.textContent).toBe("Login");
    expect(existingUserLabelElement.textContent).toBe("Existing User");

    fireEvent.change(usernameInputElement, { target: { value: "user" } });
    fireEvent.change(passwordInputElement, { target: { value: "password" } });
    expect(usernameInputElement.value).toBe("user");
    expect(passwordInputElement.value).toBe("password");
  });
});
