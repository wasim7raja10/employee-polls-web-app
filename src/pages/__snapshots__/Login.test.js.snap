// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" flex justify-center flex-col items-center gap-6"
      >
        <h1
          class=" text-blue-400 text-4xl p-8 text-center"
        >
          Login
        </h1>
        <div
          class="dropdown"
        >
          <label
            class="btn m-1"
            data-testid="existing-user-label"
            tabindex="0"
          >
            Existing User
          </label>
          <ul
            class="dropdown-content menu p-2 shadow bg-base-100 rounded-box w-52"
            tabindex="0"
          >
            <li>
              <span>
                Sarah Edo
              </span>
            </li>
            <li>
              <span>
                Tyler McGinnis
              </span>
            </li>
            <li>
              <span>
                Mike Tsamis
              </span>
            </li>
            <li>
              <span>
                Zenobia Oshikanlu
              </span>
            </li>
          </ul>
        </div>
        <form
          class=" form-control w-full max-w-xs space-y-6"
        >
          <div
            class=" space-y-2"
          >
            <label
              data-testid="username-label"
              for="username"
            >
              Username
            </label>
            <input
              class="input input-bordered w-full max-w-xs"
              data-testid="username-input"
              id="username"
              name="username"
              type="text"
              value="tylermcginnis"
            />
          </div>
          <div
            class=" space-y-2"
          >
            <label
              data-testid="password-label"
              for="password"
            >
              Password
            </label>
            <input
              class="input input-bordered w-full max-w-xs"
              data-testid="password-input"
              id="password"
              name="password"
              type="password"
              value="abc321"
            />
          </div>
          <button
            class="btn"
            data-testid="submit-login"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" flex justify-center flex-col items-center gap-6"
    >
      <h1
        class=" text-blue-400 text-4xl p-8 text-center"
      >
        Login
      </h1>
      <div
        class="dropdown"
      >
        <label
          class="btn m-1"
          data-testid="existing-user-label"
          tabindex="0"
        >
          Existing User
        </label>
        <ul
          class="dropdown-content menu p-2 shadow bg-base-100 rounded-box w-52"
          tabindex="0"
        >
          <li>
            <span>
              Sarah Edo
            </span>
          </li>
          <li>
            <span>
              Tyler McGinnis
            </span>
          </li>
          <li>
            <span>
              Mike Tsamis
            </span>
          </li>
          <li>
            <span>
              Zenobia Oshikanlu
            </span>
          </li>
        </ul>
      </div>
      <form
        class=" form-control w-full max-w-xs space-y-6"
      >
        <div
          class=" space-y-2"
        >
          <label
            data-testid="username-label"
            for="username"
          >
            Username
          </label>
          <input
            class="input input-bordered w-full max-w-xs"
            data-testid="username-input"
            id="username"
            name="username"
            type="text"
            value="tylermcginnis"
          />
        </div>
        <div
          class=" space-y-2"
        >
          <label
            data-testid="password-label"
            for="password"
          >
            Password
          </label>
          <input
            class="input input-bordered w-full max-w-xs"
            data-testid="password-input"
            id="password"
            name="password"
            type="password"
            value="abc321"
          />
        </div>
        <button
          class="btn"
          data-testid="submit-login"
        >
          Login
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
